### ISSUE TITLE
Uploading > 35k image files

### ISSUE THREAD

**Issue by jluethi** [TissueMAPS/TissueMAPS#107]
I'm trying to upload a big 3D experiment and I run into an issue during the upload. It uploaded ~35k files, then it seems to have run into an authorization timeout. When I try to rerun the upload, I get the following error after/during file registration (as extracted from the uwsgi log, the console reports a 504 server error: gateway timeout):
Thu Dec 21 09:42:36 2017 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 296] during POST /api/experiments/dG1hcHMz/acquisitions/dG1hcHMx/upload/register (172.23.177.25)
IOError: write error
Does anyone know how I can trace this?
@scottberry Have you ever seen this? I assume you have uploaded > 35k files in the past?

I'm running on v0.3.3 tm_client and the current master for TissueMaps (state 15/12/2017). My upload command was:
tm_client -vvv -H 172.23.199.185 -u USERNAME -p PASSWORD microscope-file -e 20171211-Multiplexing-Antibody-3D upload -p p1 -a a1 --directory DIRECTORY_PATH

**Comment by riccardomurri** [comment]
I would first check that you have enough space on the server to
accomodate all the 35k files.

**Comment by jluethi** [comment]
Thanks @riccardomurri 
Space luckily isn't the problem, I attached a 5TB volume and I'm currently using 11% of it. 35k files are now on there, but there are abut as many again that aren't on there yet.

**Comment by riccardomurri** [comment]
Well, then things I would check are:

1. Is the server still running? (= can you log in, list experiments,
   etc.)  If not, try to restart it

2. If the server is running, check the logs: there should be a Python
   traceback corresponding to the file upload request that failed, which
   gives a hint as to where the problem might lie.

3. It may just be a timing problem: connections to the server are
   authenticated through a "JWT token" -- tokens are time-limited and
   will expire.  If the data transfer goes on for a long time, the JWT
   token can expire in the middle of it, and the transfer fails
   (although in this case you should get an error code "403
   Unauthorized")

**Comment by jluethi** [comment]
Thanks for the ideas+
1. The server is running.
2. What logs other than the uwsgi logs can I check? The error message above is from the uwsgi log.
3. I suspect this is what happened during the first run, because it uploaded some 35k files and then failed with an authentication error. To upload the remaining images, I tried to rerun the same command. I reran this upload 3 times this afternoon and reran it now, always getting the same error.

Do you have ideas for other logs to check or other things to look at?

**Comment by riccardomurri** [comment]
> 2. What logs other than the uwsgi logs can I check? The error message
> above is from the uwsgi log.

I meant the uWSGI log. Is there any Python traceback around that error
line you quoted?

**Comment by jluethi** [comment]
No, unfortunately the statements above and below are just normal log entries about status checking.

On 21 Dec 2017, at 17:38, Riccardo Murri <notifications@github.com<mailto:notifications@github.com>> wrote:

> 2. What logs other than the uwsgi logs can I check? The error message
> above is from the uwsgi log.

I meant the uWSGI log. Is there any Python traceback around that error
line you quoted?

**Comment by riccardomurri** [comment]
> No, unfortunately the statements above and below are just normal log
> entries about status checking.

My guess then would be is that some internal buffer of uWSGI or NginX is
overflown.  Does it work if you e.g. upload one image only?  Is there a
maximum size of the image you can upload?
